src/test/java/com/github/davidmoten/rx2/aws/SqsTest.java

```diff
@@ -292,63 +314,129 @@ public void testPollingReturnsAllAvailableMessagesAtEachScheduledCall() {
+        List<Message> list = new ArrayList<>();
+        List<String> events = new ArrayList<>();
+        List<Message> result = Sqs.messages(() -> list, () -> events.add("prePoll"), e -> events.add("postPoll"));
+        assertTrue(result == list);
+        assertEquals(Arrays.asList("prePoll", "postPoll"), events);
+    }
+    
+    @Test
+    public void testPollEventsIfPrePollThrows() {
+        List<Message> list = new ArrayList<>();
+        List<String> events = new ArrayList<>();
+        try {
+            Sqs.messages(() -> list, () -> {
+                throw new RuntimeException();
+            }, e -> events.add("postPoll"));
+            Assert.fail();
+        } catch (RuntimeException e) {
+            // expected
+            assertTrue(events.isEmpty());
+        }
+    }
+    
+    @Test
+    public void testPollEventsIfSupplierThrows() {
+        List<String> events = new ArrayList<>();
+        try {
+            Sqs.messages(() -> {
+                throw new RuntimeException();
+            }, () -> events.add("prePoll"), //
+                    e -> events.add(e.get().getClass().getSimpleName()));
+            Assert.fail();
+        } catch (RuntimeException e) {
+            // expected
+            assertEquals(Arrays.asList("prePoll", "RuntimeException"), events);
+        }
+    }
+    
+    @Test
+    public void testPollEventsIfPostPollThrows() {
+        List<String> events = new ArrayList<>();
+        try {
+            Sqs.messages(() -> new ArrayList<>(), //
+                    () -> events.add("prePoll"), //
+                    e -> {
+                        throw new IllegalArgumentException();
+                    });
+            Assert.fail();
+        } catch (IllegalArgumentException e) {
+            // expected
+            assertEquals(Arrays.asList("prePoll"), events);
+        }
+    }
+    
+    @Test
+    public void testUncheckedCall() {
+        assertEquals(1, (int) Sqs.uncheckedCall(() -> 1));   
+    }
+    
+    @Test
+    public void testUncheckedCallThrowsRuntimeException() {
+        try {
+            Sqs.uncheckedCall(() -> {
+                throw new IllegalArgumentException("boo");
+            });
+            Assert.fail();
+        } catch (RuntimeException e) {
+            assertEquals("boo", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void testUncheckedCallThrowsError() {
+        try {
+            Sqs.uncheckedCall(() -> {
+                throw new Error("boo");
+            });
+            Assert.fail();
+        } catch (Error e) {
+            assertEquals("boo", e.getMessage());
+        }
+    }
+    
+    @Test
+    public void testUncheckedCallThrowsCheckedException() {
+        try {
+            Sqs.uncheckedCall(() -> {
+                throw new Exception("boo");
+            });
+            Assert.fail();
+        } catch (RuntimeException e) {
+            assertEquals("boo", e.getCause().getMessage());
```

Comment:
Same again - @Test(expected = RuntimeException.class)

Reply:
this pattern is used because another assertion happens after exception thrown